backend/modules/professor/routers/professor_grading.py

```
@@ -417,3 +418,173 @@ async def revert_to_ai_grade(
+
+        # Get the simulation to check ownership
+        simulation = db.query(Scenario).filter(
+            Scenario.id == user_progress.simulation_id
+        ).first()
+
+        if not simulation:
+            raise HTTPException(status_code=404, detail="Simulation not found")
+
+        # Check if professor owns the simulation or is admin
+        if current_user.role != "admin" and simulation.created_by != current_user.id:
+            raise HTTPException(status_code=403, detail="Not authorized to regrade this simulation")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)